integration_tests/test_game.py

```diff
@@ -0,0 +1,106 @@
+import asyncio
+
+import pytest
+
+# All test coroutines will be treated as marked.
+pytestmark = pytest.mark.asyncio
+
+
+async def simulate_game(host, *guests, results=[]):
+    all_clients = [host] + list(guests)
+    game_id = await host.host_game()
+    await host.configure_joining_player(host.player_id, 1)
+    for guest in guests:
+        await guest.join_game(game_id)
+        await host.configure_joining_player(guest.player_id, 2)
+
+    await host.send_message({
+        "target": "game",
+        "command": "GameState",
+        "args": ["Launching"]
+    })
+    for client in all_clients:
+        await client.read_until(
+            lambda msg: (
+                msg.get("command") == "game_info" and
+                msg["host"] == "test" and
+                msg["launched_at"] is not None
+            )
+        )
+
+    await host.send_message({
+        "target": "game",
+        "command": "EnforceRating"
+    })
+
+    for client in all_clients:
+        for result in results:
+            await client.send_message({
+                "target": "game",
+                "command": "GameResult",
+                "args": result
+            })
+
+    # Report GameEnded
```

Comment:
Is this what's send when I click "exit to score" or the like?

Reply:
I assume these are meant as cleanup for the test server rather than as part of the test, since it happens after all assertions? It would be nice to put this in some kind of teardown part so that they disconnect and the game on the test server is cleaned up even if the test fails. (Although I'm not completely sure how to have teardown for anything in pytest that's not a fixture right now.)

Reply:
These are reported as soon as the simulation ends, so when you see "Game Over" appear on the screen. No need to click on the score button.